pom.xml

```diff
@@ -96,19 +96,24 @@ under the License.
     <dependency>
       <groupId>org.eclipse.sisu</groupId>
       <artifactId>org.eclipse.sisu.plexus</artifactId>
-      <version>0.3.3</version>
+      <version>0.3.4</version>
       <exclusions>
         <exclusion>
           <groupId>org.sonatype.sisu</groupId>
           <artifactId>sisu-guice</artifactId>
         </exclusion>
+        <exclusion>
+          <groupId>org.codehaus.plexus</groupId>
+          <artifactId>plexus-utils</artifactId>
+        </exclusion>
       </exclusions>
     </dependency>
 
     <dependency>
       <groupId>com.google.inject</groupId>
       <artifactId>guice</artifactId>
-      <version>4.0</version>
+      <version>4.2.1</version>
+      <classifier>no_aop</classifier>
```

Comment:
Thanks for your time reviewing it. Just in case, the reason I updated this version is because Maven 3.8.1 depends on it. It doesn't necessarily mean it's a proper one in this case, just wanted to mention where it came from.

Reply:
> Maven 3.8.1 depends on it

Or claims to. Regardless, unless you take special steps, code loaded inside the Jenkins controller or agent is going to used the 4.0 bundled in core.

Reply:
No problem. Thanks.

Reply:
> the 4.0 bundled in core

As of https://github.com/jenkinsci/jenkins/pull/5858 this will be 5.x. The dependency should be deleted, and if necessary `exclude`d from some other `dependency`.

Probably ought to have a `RealJenkinsRule` test verifying that this upgrade breaks nothing, and try on a recently weekly.